### PR TITLE
Json config parser

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add a json parser to the CLI11 parser and use it for parsing a config file in the FederateInfo.  This is a step toward getting the multibroker operational.  We needed a better way to handle config files for the different core and network info.  This is one step on that path.  There will be a couple more coming.  But I thought it useful to break up the process a little.  This includes several unreleased modifications to CLI11 that I will be working upstream in the near future.  

### Proposed changes

- update CLI11 to master+some coming additions 
- use CLI11 to handle the parsing of config files from toml and JSON for command line like arguments.  
-  some tweaks to the config file handling for FederateInfo Objects. 

Next steps will be some further work on handling config files for core configurations.

### CODE REVIEW

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ src/helics/application_api/FederateInfo.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5330956&bid=17447660)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ src/helics/application_api/FederateInfo.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5330956&bid=17447660)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ src/helics/application_api/FederateInfo.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5330956&bid=17447660)

File: ThirdParty/helics/external/CLI11/CLI11.hpp [hunk 1]
```diff
@@ -5233,6 +5734,54 @@ class App {
...
+                                             [vfunc]() { throw(CLI::CallForVersion(vfunc(), 0)); },
+                                             "display program version information and exit");
+            version_ptr_->configurable(false);
+        }
+
+        return version_ptr_;
+    }
+    /// Set the version as a string
+    App *version(const std::string &version) {
+        set_version_flag("-v,--version", version);
```
Comment (nightlark) [inline ThirdParty/helics/external/CLI11/CLI11.hpp:5775]: I'm don't think `-v` should be "reserved" by default. It's used for verbose by a lot of commonly used commands (rm, cp, mv, curl, wget).

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ src/helics/application_api/FederateInfo.cpp  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5330956&bid=17447660)

**Pull request merged**